### ISSUE TITLE
Upgrade `aria-query` to version 5.2.1

### ISSUE THREAD

**Issue by MatanBobi** [testing-library/dom-testing-library#1239]
### Describe the feature you'd like:

`aria-query` has released a new version yesterday which caused some regression on our end.
At the moment, we've pinned the version to prevent issues, we need to upgrade it soon to be compliant with the new ARIA spec changes.
Affected issues that were closed in the meantime because we've pinned the version:
https://github.com/testing-library/dom-testing-library/issues/1235
https://github.com/testing-library/dom-testing-library/issues/1238

### Teachability, Documentation, Adoption, Migration Strategy:

We need to upgrade locally, see which tests are failing and why, if `aria-query` has issues, we need to open issues in their repo.

**Comment by philer-jambit** [comment]
The current pin (`aria-query: 5.1.3`) causes some of our tests to fail, in particular:

* an `<input type"text" list="my-options" … />` which references a `<datalist id="my-options">` is no longer accessible as `role="textbox"` but `role="combobox"` instead
* an `<input type="date" />` which was previously accessible as `role="textbox"` now has no role at all.

Note that we are using `@testing-library/cypress`, which still uses the outdated `@testing-library/dom` version `^8.1.0`.

**Comment by MatanBobi** [comment]
Hi @philer-jambit, thanks for posting this.
The behavior you're seeing in 5.1.3 is correct and follows the spec so you'll probably need to adapt your tests:
1. An input with `type="text"` and a `list` attribute should be accessible through `role="combobox"`:
<img width="431" alt="image" src="https://github.com/testing-library/dom-testing-library/assets/12711091/bb45e46d-2721-4581-9db7-c71b207e94eb">
2. Input `type="date"` has no corresponding implicit role:
<img width="397" alt="image" src="https://github.com/testing-library/dom-testing-library/assets/12711091/d6fea6cd-f285-4bd2-80d7-3d2c1779f00c">

More about this can be found in the [ARIA spec](https://www.w3.org/TR/html-aria/).
Thanks again!

**Comment by philer-jambit** [comment]
Good to know, thanks! :+1: 

I have to say, date inputs not having any role of their own and the spec forbidding adding a role is quite impractical. Firefox's accessibility inspector actually calls it `date editor`, but evidently that is non-standard. For lack of a better idea I've now resorted to `data-testid`.

**Comment by cmorten** [comment]
A new version https://www.npmjs.com/package/aria-query/v/5.3.0, it would be good to see if this resolves the issues with `5.2.1`
